### ISSUE TITLE
Handle exceptions on make_api_call

### ISSUE THREAD

**Issue by jletizia** [jletizia/dashboardapi#3]
Currently `make_api_call` is allowing it's call to HTTParty to handle all exceptions. More robust exception handling to provide more relevant data to the actual API should be added.

**Comment by jletizia** [comment]
POST will now return the exact error if Dashboard returned one (#6)

```ruby
def make_api_call(endpoint_url, http_method, options_hash={})
    headers = {"X-Cisco-Meraki-API-Key" => @key}
    headers.merge!(options_hash[:headers]) if options_hash[:headers]

    options = {:headers => headers, :body => options_hash[:body].to_json}
    case http_method
    when 'GET'
      res = HTTParty.get("#{self.class.base_uri}/#{endpoint_url}", options)
      return JSON.parse(res.body)
    when 'POST'
      res = HTTParty.post("#{self.class.base_uri}/#{endpoint_url}", options)
      raise "Bad Request due to the following error(s): #{res['errors']}" if res['errors']
      return res
    end

  end
```

**Comment by jletizia** [comment]
this is resolved in #7. All 4 HTTP METHODS now check:

```ruby
raise "Bad Request due to the following error(s): #{res['errors']}" if res['errors']
raise "404 returned. Are you sure you are using the proper IDs?" if res.code == 404
```
